compute_tools/src/bin/compute_ctl.rs

```diff
@@ -275,12 +276,77 @@ fn main() -> Result<()> {
         "running compute with features: {:?}",
         state.pspec.as_ref().unwrap().spec.features
     );
+    // before we release the mutex, fetch the swap size (if any) for later.
+    let swap_size_bytes = state.pspec.as_ref().unwrap().spec.swap_size_bytes;
     drop(state);
 
     // Launch remaining service threads
     let _monitor_handle = launch_monitor(&compute);
     let _configurator_handle = launch_configurator(&compute);
 
+    // Resize swap to the desired size if the compute spec says so
+    if let (Some(size_bytes), true) = (swap_size_bytes, resize_swap_on_bind) {
```

Comment:
This feels _maybe_ like a very large block of code to insert into the body of this function.

Maybe it should go in a separate file? (e.g., similar to the function of `compute_tools/src/configurator.rs`).

Curious for thoughts

Reply:
Moved to `compute_tools/src/swap.rs`